nixos/tests/installer.nix

```diff
@@ -110,8 +126,45 @@ let
       def create_machine_named(name):
           return create_machine({**default_flags, "name": name})
 
+      class Tpm:
+            def __init__(self):
+                self.start()
+
+            def start(self):
+                self.proc = subprocess.Popen(["${pkgs.swtpm}/bin/swtpm",
+                    "socket",
+                    "--tpmstate", "dir=/tmp/swtpm",
+                    "--ctrl", "type=unixio,path=/tmp/swtpm-sock",
+                    "--tpm2"
+                    ])
+
+                # Check whether starting swtpm failed
+                try:
+                    exit_code = self.proc.wait(timeout=0.2)
+                    if exit_code is not None and exit_code != 0:
+                        raise Exception("failed to start swtpm")
+                except subprocess.TimeoutExpired:
+                    pass
+
+            """Check whether the swtpm process exited due to an error"""
+            def check(self):
+                exit_code = self.proc.poll()
+                if exit_code is not None and exit_code != 0:
+                    raise Exception("swtpm process died")
+
+
+      os.mkdir("/tmp/swtpm")
```

Comment:
Don't use `/tmp`. Use `$NIX_BUILD_TOP`, or a `mktemp`, or something like that.

Reply:
Is that really possible ? The same path has to be provided under `virtualisation.qemu.options` in the installer config.

Reply:
I mean it's a python script. You can put a temporary directory path or the value of the `NIX_BUILD_TOP` env variable into a python variable and use that in strings. It's not a good idea to use hardcoded paths in `/tmp` because the sandbox may not necessarily be enabled.

Reply:
In the python script yes, but the same directory has to be provided here: https://github.com/NixOS/nixpkgs/blob/e422cd241f31b38c5613579388a0ec3b1a64c276/nixos/tests/installer.nix#L482

Reply:
Hm. Well, as a quick workaround, I'm pretty sure that `qemu.options` isn't shell-escaped, which lets you use environment variables there. It's not great but it's better than hard coding `/tmp`

Reply:
Oh yes you are right this is working :) Fixed